### PR TITLE
Allow `undefined` onBarCodeScanned prop

### PR DESCRIPTION
**Pull request by josmithua**
# Why

Currently this line in the example code in the official docs produces a type error: 
```
onBarCodeScanned={code ? undefined : handleBarCodeScanned}
``` 
because the prop doesn't expect `undefined`. Yet it's a valid, documented value that enables the user to pause scanning. 

# How

This PR changes the type of the onBarCodeScanned prop to allow `undefined`

# Test Plan

1. Create a managed typescript expo project
1. `expo install expo-barcode-scanner`
1. Copy the official example from the docs (https://docs.expo.io/versions/latest/sdk/bar-code-scanner/)
1. Notice the type error as described above.
1. Pull this PR, and notice no more error.
1. yarn build successful

### CODE REVIEW

**Review by lukmccall — CHANGES_REQUESTED** [review]
Thanks for contributing 🎉🎉🎉  Awesome work ;)
However, you need to do two things:
- run `yarn build` in `expo-barcode-scanner` directory
- add an entry to changelog -> https://github.com/expo/expo/blob/master/packages/expo-barcode-scanner/CHANGELOG.md

**Comment by josmithua** [comment]
@lukmccall I've updated the CHANGELOG, and built the package (builds fine).

Cheers

File: packages/expo-barcode-scanner/CHANGELOG.md [hunk 1]
```diff
@@ -8,6 +8,8 @@
 
 ### 🐛 Bug fixes
 
+- Allow `onBarCodeScanned` prop to be `undefined`
```
Comment (lukmccall) [inline packages/expo-barcode-scanner/CHANGELOG.md:11]:
```suggestion
- Allow `onBarCodeScanned` prop to be `undefined`. ([#10068](https://github.com/expo/expo/pull/10068) by [@josmithua](https://github.com/josmithua))
```

**Review by lukmccall — CHANGES_REQUESTED** [review]

**Comment by lukmccall** [comment]
Please, add rebuild files to this PR ;)

**Comment by josmithua** [comment]
Sorry, not sure how I missed the changed build files before!

**Review by lukmccall — APPROVED** [review]
LGTM 🎉🎉🎉

**Pull request merged**